op-node/rollup/driver/state.go

```diff
@@ -556,8 +561,8 @@ type hashAndErrorChannel struct {
 // WARNING: This is only an outgoing signal, the blocks are not guaranteed to be retrieved.
 // Results are received through OnUnsafeL2Payload.
 func (s *Driver) checkForGapInUnsafeQueue(ctx context.Context) error {
-	start := s.derivation.UnsafeL2Head()
-	end := s.derivation.UnsafeL2SyncTarget()
+	start := s.engineController.UnsafeL2Head()
+	end := s.derivation.HighestUnsafeBlock()
```

Comment:
I think this is what's breaking the P2P alt sync tests. Previously `UnsafeL2SyncTarget` was the lowest pending unsafe block, allowing this function to calculate the gap between the current L2 unsafe head and the next unsafe block we had to import.  Now that it's the highest block we are trying to rerequest all the pending blocks we already have rather than just filling the gap.

Reply:
that seems like it'd be it. I'm dumb & was kinda randomly patching stuff up. This also likely breaks the assumption of what syncing was supposed to do.

Reply:
Fixed this. It also simplified things which was nice.